cpp/src_prims/random/rng.cuh

```diff
@@ -128,7 +130,7 @@ class Rng {
       offset(0),
       // simple heuristic to make sure all SMs will be occupied properly
       // and also not too many initialization calls will be made by each thread
-      nBlocks(4 * getMultiProcessorCount()),
+      nBlocks(4 * MLCommon::getMultiProcessorCount()),
```

Comment:
When rng.cuh moves it RAFT, it means we'll still have a raft->cuml circular dependency, right?

Reply:
I don't think we'd want to introduce that by any means. My thinking was that we'll keep two versions of util files: one in cuml codebase with cuml namespaces, and a mirror one in RAFT. So, we move `src_prims/cuda_utils.cuh` to RAFT also

Reply:
the lesser of 2 evils being some rather small code duplication, which I definitely agree with